File: lab-1-ByLiza/с++/trapezoid.cpp
#include <iostream>
#include "trapezoid.h"
#include <cmath>

Trapezoid::Trapezoid()
{
    x1 = y1 = x2 = y2 = x3 = y3 = x4 = y4 = 0.0;
}

Trapezoid::Trapezoid(double x1, double y1, double x2, double y2, double x3, double y3, double x4, double y4)
{
    this->x1 = x1;
    this->y1 = y1;
    this->x2 = x2;
    this->y2 = y2;
    this->x3 = x3;
    this->y3 = y3;
    this->x4 = x4;
    this->y4 = y4;
}

Trapezoid::Trapezoid(const Trapezoid &other)
{
    x1 = other.x1;
    y1 = other.y1;
    x2 = other.x2;
    y2 = other.y2;
    x3 = other.x3;
    y3 = other.y3;
    x4 = other.x4;
    y4 = other.y4;
}

Trapezoid::~Trapezoid()
{
    std::cout << "Трапеція знищена" << std::endl;
}

double Trapezoid::getX1() const { return x1; }
double Trapezoid::getY1() const { return y1; }
double Trapezoid::getX2() const { return x2; }
double Trapezoid::getY2() const { return y2; }
double Trapezoid::getX3() const { return x3; }
double Trapezoid::getY3() const { return y3; }
double Trapezoid::getX4() const { return x4; }
double Trapezoid::getY4() const { return y4; }

double Trapezoid::getPerimeter()
{
    double sideA = std::sqrt((x2 - x1) * (x2 - x1) + (y2 - y1) * (y2 - y1));
    double sideB = std::sqrt((x4 - x3) * (x4 - x3) + (y4 - y3) * (y4 - y3));
    double sideC = std::sqrt((x3 - x2) * (x3 - x2) + (y3 - y2) * (y3 - y2));
    double sideD = std::sqrt((x4 - x1) * (x4 - x1) + (y4 - y1) * (y4 - y1));

    return sideA + sideB + sideC + sideD;
}

double Trapezoid::getArea()
{
    double baseA = std::sqrt((x2 - x1) * (x2 - x1) + (y2 - y1) * (y2 - y1));
    double baseB = std::sqrt((x4 - x3) * (x4 - x3) + (y4 - y3) * (y4 - y3));
    double height = std::abs(y3 - y1);
    return (baseA + baseB) * height / 2.0;
}

File: lab-1-ByLiza/c#/Trapezoid.cs
namespace TrapezoidApp
{
    public class Trapezoid
    {
        private double _x1, _y1;
        private double _x2, _y2;
        private double _x3, _y3;
        private double _x4, _y4;

        public double X1 { get; private set; }
        public double Y1 { get; private set; }
        public double X2 { get; private set; }
        public double Y2 { get; private set; }
        public double X3 { get; private set; }
        public double Y3 { get; private set; }
        public double X4 { get; private set; }
        public double Y4 { get; private set; }

        public Trapezoid()
        {
            _x1 = _y1 = _x2 = _y2 = _x3 = _y3 = _x4 = _y4 = 0.0;
        }

        public Trapezoid(double x1, double y1, double x2, double y2, double x3, double y3, double x4, double y4)
        {
            _x1 = x1;
            _y1 = y1;
            _x2 = x2;
            _y2 = y2;
            _x3 = x3;
            _y3 = y3;
            _x4 = x4;
            _y4 = y4;
        }
        public Trapezoid(Trapezoid other)
        {
            _x1 = other._x1;
            _y1 = other._y1;
            _x2 = other._x2;
            _y2 = other._y2;
            _x3 = other._x3;
            _y3 = other._y3;
            _x4 = other._x4;
            _y4 = other._y4;
        }
        ~Trapezoid()
        {
            Console.WriteLine("Об'єкт Trapezoid знищено.");
        }

        public double GetPerimeter()
        {
            double sideA = Math.Sqrt((_x2 - _x1) * (_x2 - _x1) + (_y2 - _y1) * (_y2 - _y1));
            double sideB = Math.Sqrt((_x4 - _x3) * (_x4 - _x3) + (_y4 - _y3) * (_y4 - _y3));
            double sideC = Math.Sqrt((_x3 - _x2) * (_x3 - _x2) + (_y3 - _y2) * (_y3 - _y2));
            double sideD = Math.Sqrt((_x4 - _x1) * (_x4 - _x1) + (_y4 - _y1) * (_y4 - _y1));

            return sideA + sideB + sideC + sideD;
        }

        public double GetArea()
        {
            double baseA = Math.Sqrt((_x2 - _x1) * (_x2 - _x1) + (_y2 - _y1) * (_y2 - _y1));
            double baseB = Math.Sqrt((_x4 - _x3) * (_x4 - _x3) + (_y4 - _y3) * (_y4 - _y3));
            double height = Math.Abs(_y3 - _y1);
            return (baseA + baseB) * height / 2.0;
        }
    }
}